src/util/address.rs

```diff
@@ -73,15 +75,14 @@ impl fmt::Display for Error {
             Error::Bech32(ref e) => write!(f, "bech32: {}", e),
             Error::EmptyBech32Payload => write!(f, "the bech32 payload was empty"),
             Error::InvalidWitnessVersion(v) => write!(f, "invalid witness script version: {}", v),
-            Error::InvalidWitnessProgramLength(l) => write!(
-                f,
-                "the witness program must be between 2 and 40 bytes in length: lengh={}",
-                l
+            Error::InvalidWitnessProgramLength(l) => write!(f,
+                "the witness program must be between 2 and 40 bytes in length: lengh={}", l,
```

Comment:
Off topic, but maybe fix this `lengh={}`?

Reply:
Oh yeah didn't notice. I can add a commit for that.

And yeah agreed that it's a bit sad it returns an error. I hope the explicit documentation of the error case can make people comfortable enough to just `unwrap` it.